sprints/dashboard/helpers.py

```diff
@@ -0,0 +1,366 @@
+            return 0
+
+    def calculate_review_time(self) -> int:
+        """
+        Calculate time needed for the review.
+        Unless directly specified (with Jira bot directive), we're planning 2 hours for stories bigger than 3 points.
+        """
+        planned = self.get_bot_directive(SPRINT_REVIEW_DIRECTIVE)
+        if planned:
+            return planned * SECONDS_IN_HOUR
+
+        if self.story_points <= 3:
+            return SECONDS_IN_HOUR
+        return 2 * SECONDS_IN_HOUR
+
+    def get_recurring_time(self) -> int:
+        """Get required assignee time for the recurring story."""
+        planned = self.get_bot_directive(SPRINT_RECURRING_DIRECTIVE)
+        return planned * SECONDS_IN_HOUR
+
+    def get_epic_management_time(self) -> int:
+        """Get required assignee time for managing the epic."""
+        planned = self.get_bot_directive(SPRINT_EPIC_DIRECTIVE)
+        return planned * SECONDS_IN_HOUR or SPRINT_HOURS_RESERVED_FOR_EPIC_MANAGEMENT * SECONDS_IN_HOUR
+
+
+class DashboardRow:
+    """Represents single dashboard row (user)."""
+
+    def __init__(self, user: Union[JiraUser, str, None]) -> None:
+        super().__init__()
+        self.user = user
+        try:
+            self.local_user = User.objects.get(email=self.user.emailAddress)
+        except (User.DoesNotExist, AttributeError):
+            self.local_user = None
+        self.current_remaining_assignee_time = 0
+        self.current_remaining_review_time = 0
+        self.current_remaining_upstream_time = 0
+        self.future_remaining_assignee_time = 0
+        self.future_remaining_review_time = 0
+        self.future_epic_management_time = 0
+        self.goal_time = 0
+        self.current_invalid = []
+        self.future_invalid = []
+
+    def set_goal_time(self, goal) -> None:
+        """
+        Calculate goal time for this user.
+        Use default if time not specified in user's profile.
+        """
+        self.goal_time = goal - SPRINT_HOURS_RESERVED_FOR_MEETINGS * SECONDS_IN_HOUR
+
+    @property
+    def committed_time(self) -> float:
+        """Calculate summary time for the upcoming sprint."""
+        return self.current_remaining_assignee_time \
+               + self.current_remaining_review_time \
+               + self.current_remaining_upstream_time \
+               + self.future_remaining_assignee_time \
+               + self.future_remaining_review_time \
+               + self.future_epic_management_time
+
+    @property
+    def remaining_time(self) -> float:
+        """Calculate available time for the upcoming sprint."""
+        return self.goal_time - self.committed_time
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, JiraUser):
+            return self.user == other
+        return self.user == other.user
+
+    def add_invalid_issue(self, issue: DashboardIssue) -> None:
+        """Add non-estimated issue to the list of invalid issues."""
```

Comment:
@Agrendalath, nit: is `invalid` the correct word? It is just issues with 0 time remaining.

Reply:
@lgp171188, would `Unestimated` (or `Non-estimated`) be a better name for this method and for the related column on the dashboard?

Reply:
@Agrendalath, `unestimated`, `non-estimated` will not be appropriate for issues with 0 remaining time.